packages/apps/composer-app/src/functions/_worker.ts

```diff
@@ -8,9 +8,40 @@
  * Output _worker.js to <pages_build_output_dir> and deploy via git.
  */
 const handler: ExportedHandler<{ ASSETS: Fetcher }> = {
-  // NOTE: This is just a pass-through. Leaving the worker setup here for future use.
   fetch: async (request, env, _context) => {
-    return env.ASSETS.fetch(request);
+    const assetsPromise = env.ASSETS.fetch(request);
+    let telemetryPromise: Promise<Response> | undefined;
+
+    // Emit telemetry log to Grafana Loki cloud.
+    if (env.DX_LOKI_ENDPOINT && env.DX_LOKI_AUTHORIZATION && request.url && request.url.includes('site.webmanifest')) {
+      telemetryPromise = fetch(env.DX_LOKI_ENDPOINT, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: env.DX_LOKI_AUTHORIZATION
+        },
+        body: JSON.stringify({ 
+          streams: [
+            {
+              stream: {
+                "service_name": "composer",
+                "environment": env.ENVIRONMENT ?? undefined
+              },
+              values: [[String(Date.now() * 1e6), "Fetch app"]]
+            }
+          ] 
+        })
+      }).then(async (response) => {
+        if (!response.ok) {
+          console.log("LOKI telemetry fetch failed:", await response.text());
+        }
+      }).catch((error) => console.error("LOKI telemetry fetch error:", error));
+    }
+
+    // Wait for both promises to complete.
+    await Promise.all([assetsPromise, telemetryPromise]);
```

Comment:
does it need to wait for the telemetry promise to resolve before serving the assets? The request for assets shouldn't fail just because of an issue with loki

Reply:
It is necessary, I tested it locally, and if it is not awaited the event is not propagated to grafana, I can add timeout on `telemetryPromise`